src/riak_kv_qry_buffers.erl

```diff
@@ -185,13 +219,10 @@ offset_to_scalar([A]) when is_integer(A), A >= 0 -> A.
           %% total chunks needed (== number of subqueries)
           chunks_need :: integer(),
 
-          %% flipped to true when chunks_need == chunks_got
-          all_chunks_received = false :: boolean(),
-
           %% total records stored (when complete)
           total_records = 0 :: non_neg_integer(),
 
-          %% %% iterator cache (need a working iterator support in eleveldb)
+          %% TODO: iterator cache
```

Comment:
The lifecycle of each qbuf table is such that after writing to it is completed, it becomes a read-only. table (effectively, the last snapshot by your definition?). This is when I intended to open some number of iterators, with the idea to have a map of `{Pos, Iter}`, so that when I need to fulfill a query with OFFSET N, I could pick the iterator whose associated `Pos` is closest to `N` and use it instead of iterating from the beginning.

However, there is no pressing need to develop on this further as query buffers are currently not reusable.